src/main/java/ca/teamdman/sfm/common/blockentity/ManagerBlockEntity.java

```diff
@@ -157,32 +158,48 @@ public int getUnprocessedRedstonePulseCount() {
     }
 
     public State getState() {
-        if (getDisk().isEmpty()) return State.NO_DISK;
-        if (getProgramString().isEmpty()) return State.NO_PROGRAM;
+        if (getDisk() == null) return State.NO_DISK;
+        if (getProgramString() == null) return State.NO_PROGRAM;
         if (program == null) return State.INVALID_PROGRAM;
         return State.RUNNING;
     }
 
-    public Optional<String> getProgramString() {
-        return getDisk().map(DiskItem::getProgram).filter(prog -> !prog.isBlank());
+    @Nullable
+    public String getProgramString() {
+        var disk = getDisk();
+        if (disk == null) {
+            return null;
+        }
+
+        var program = DiskItem.getProgram(disk);
+        return program.isBlank() ? null : program;
+    }
+
+    public String getProgramStringOrEmptyIfNull() {
+        var programString = this.getProgramString();
+        return programString == null ? "" : programString;
     }
 
     public Set<String> getReferencedLabels() {
         if (program == null) return Collections.emptySet();
         return program.referencedLabels();
     }
 
-    public Optional<ItemStack> getDisk() {
+    @Nullable
+    public ItemStack getDisk() {
```

Comment:
```suggestion
    public @Nullable ItemStack getDisk() {
```
may be more idiomatic to return an empty stack here but for now a null is fine